### PR TITLE
Recipe app

### PR DESCRIPTION
**Pull request by Bornittah**
**Implementation**

- A login page

- A registration page.
- A food list (with all CRUD implementation, except for 'update').
- A recipes list (with all CRUD implementation, except for 'update').
- A general shopping list view (all missing food for all your recipes and total price).
- A list of all public recipes from other users with their names and total prices.

### CODE REVIEW

File: spec/features/login_spec.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
+
+RSpec.describe 'User signs in', type: :system do
+  before do
+    @user = create :user
+    visit new_user_session_path
+  end
+
+  scenario 'valid with correct credentials' do
```
Comment (danielufeli) [inline spec/features/login_spec.rb:9]: - [ ] Nice work writing tests for your project, kindly check,  your tests are failing, see the screenshot below.
error-test

File: app/controllers/foods_controller.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        format.json { render :show, status: :ok, location: @food }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @food.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /foods/1 or /foods/1.json
+  def destroy
```
Comment (danielufeli) [inline app/controllers/foods_controller.rb:47]: - [ ] Nice work done on your project, kindly check when I click the delete button on the food page it throws the error below.
error-food

File: app/controllers/recipe_foods_controller.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    respond_to do |format|
+      if @recipe_food.update(recipe_food_params)
+        format.html { redirect_to recipe_url(@recipe_food.recipe_id), notice: 'Recipe food was successfully updated.' }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def destroy
```
Comment (danielufeli) [inline app/controllers/recipe_foods_controller.rb:44]: - [ ] Nicely done, kindly check when I click the delete button on the recipe page it throws the error in the screenshot below.
error-receeipe

**Review by danielufeli — CHANGES_REQUESTED** [review]
Hi @Team, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

AlmostThereReachGIF (2)

## Highlights
- Nice commits messages✔️
- Clean Code✔️

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by zainsadaqat — APPROVED** [review]
# Pull Request Approved ✅ 🎊 🎉 
## Hello Team! 👋 

## Things to Appreciate 
- [ ] Correct Flow ✅ 
- [ ] Clean Code 👍 
- [ ] No Linter Errors 🔥 
- [ ] Descriptive Pull Request and Summary 👌 

### Your project is complete!  There is nothing else to say other than it's time to merge it.

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 ☑️ 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**